Extractor/Config/ConfigToggles.cs

```diff
@@ -0,0 +1,58 @@
+namespace Cognite.OpcUa.Config
+{
+    /// <summary>
+    /// Class wrapping a config instance, enabling features that may be enabled from different places.
+    /// </summary>
+    public class ConfigToggles
+    {
+        private readonly FullConfig config;
+        public ConfigToggles(FullConfig config)
+        {
+            this.config = config;
+        }
+
+        private bool FdmEnabled => config.Cognite?.MetadataTargets?.DataModels?.Enabled ?? false;
+
+        /// <summary>
+        /// Should the extractor load references to and from properties?
+        /// </summary>
+        public bool MapPropertyReferences => FdmEnabled;
+
+        /// <summary>
+        /// Should the extractor get references to types when loading non-hierarchical references?
+        /// </summary>
+        public bool GetNonHierarchicalTypeReferences => FdmEnabled || config.Extraction.NodeTypes.AsNodes;
+        /// <summary>
+        /// Should the extractor create unmapped nodes it encounters when reading non-hierarchical references, or just ignore them?
+        /// </summary>
+        public bool CreateUnknownReferencedNodes => FdmEnabled || config.Extraction.Relationships.CreateReferencedNodes;
+        /// <summary>
+        /// Should the extractor create nodes with NodeClass ObjectType, ReferenceType, VariableType, or DataType?
+        /// </summary>
+        public bool LoadTypesAsNodes => FdmEnabled || config.Extraction.NodeTypes.AsNodes;
+
+        // Type manager
+        /// <summary>
+        /// Should the type manager load data types when reading the type hierarchy?
+        /// </summary>
+        public bool LoadDataTypes => FdmEnabled || config.Extraction.DataTypes.AutoIdentifyTypes;
+        /// <summary>
+        /// Should the type manager load event types when reading the type hierarchy?
+        /// Since event types are also type definitions, this is enabled by default if
+        /// LoadTypeDefinitions is true.
```

Comment:
This refers to some external logic. Perhaps we should exclude this information from here

Reply:
Sure

Reply:
Actually, this doesn't refer to external logic. It refers to the structure of the base OPC-UA type hierarchy, so it is inherent to OPC-UA.

The type definition hierarchy looks like this:

 - ObjectTypes
   - EventTypes
 - VariableTypes

so if type definitions are enabled, we will get event types as well.